glam/api/views.py

```diff
@@ -439,3 +443,71 @@ def random_probes(request):
         probes.append({"data": agg.histogram, "info": probe.info})
 
     return Response({"probes": probes})
+
+
+def log_probe_query(request):
+    query = request.data["query"]
+    UsageInstrumentation(
+        action_type=UsageInstrumentation.ACTION_PROBE_SEARCH,
+        context=query,
+        tracking_id=request.COOKIES.get("session"),
+        probe_name=query["probe"],
+    ).save()
+
+
+@api_view(["GET"])
+def usage(request):
+    """
+    Provides individual or aggregated (count) metrics on probe searching.
+
+    Possible query parameters are:
+    * fromDate: Date to start the search with.  Format: YYYYMMDD
+    * toDate: Date to end the search with.  Format: YYYYMMDD
+    * fields: Name of fields to return. See models.UsageInstrumentation for the full list.
+              This parameter is needed for aggregation.
+    * actionType: The type of action that triggered the metric. The only possible value now is: PROBE_SEARCH
+    * agg: The "Aggregate" flag. The only possible value now is: count. Note that if "fields" is not
+           supplied, this parameter is ignored
+    """
+    if request.method == "GET":
+        q_action_type = request.GET.get("actionType", "")
+        q_from = request.GET.get("fromDate", "")
+        q_to = request.GET.get("toDate", "")
+        q_fields = request.GET.get("fields", "")
+        q_count = request.GET.get("agg", "") == "count"
+
+        dimensions = []
+
+        if q_action_type:
+            dimensions.append(Q(action_type=q_action_type))
+        if q_from:
+            min_date = dateutil.parser.parse(q_from)
+            dimensions.append(Q(timestamp__gte=min_date))
+        if q_to:
+            max_date = dateutil.parser.parse(q_to)
+            dimensions.append(Q(timestamp__lte=max_date))
+
+        result = UsageInstrumentation.objects.filter(*dimensions)
+
+        if q_fields:
+            fields = q_fields.split(",")
+            response = result.values(*fields)
+            if q_count:
```

Comment:
Maybe if q_count is not specified we could limit the amount of rows. Just not sure what a good number is. 1k?